ios/ElectrodeApiImpl/APIImpls/Navigation/ENBaseNavigationController.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  ENBaseNavigationController.swift
+//  ElectrodeApiImpl
+//
+//  Created by Jeffrey Wang on 3/17/21.
+//  Copyright © 2021 Walmart. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class ENBaseNavigationController: UINavigationController, ENMiniAppNavDataProvider {
+
+    var rootComponentName: String
+    var properties: [AnyHashable : Any]?
+    var finish: Payload?
+    var navigateWithRoute: NavigateWithRoute = { _ in
+        return false
+    }
+    var backToMiniApp: BackToRoute = { _, _ in
+        return false
+    }
+
+    init(properties: [AnyHashable : Any]?, rootComponentName: String) {
+        self.rootComponentName = rootComponentName
+        self.properties = properties
+        super.init(nibName: nil, bundle: nil)
+        self.finish = { payload in
+            self.finishHandler(payload)
+        }
+        self.navigateWithRoute = { route in
+            return self.navigateWithRouteHandler(route)
+        }
+        self.backToMiniApp = { componentName, backProperties in
+            return self.backToMiniAppHandler(componentName, backProperties)
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let navDelegate = ENNavigationDelegate()
+        navDelegate.viewDidLoad(viewController: self)
+        addCloseButton()
+    }
+
+    func finishHandler(_ payload: [AnyHashable: Any]?) {
+        dismiss(animated: true)
+    }
+
+    func navigateWithRouteHandler(_ route: [AnyHashable: Any]) -> Bool {
```

Comment:
Can we rename this to `navigate` and pass two params (path:String and Data: Dictionary) by unwrapping `Route` object inside delegate ? Refer [Android](https://github.com/electrode-io/ernnavigation-api-impl-native/blob/master/android/lib/src/main/java/com/ern/api/impl/navigation/ElectrodeBaseActivity.java#L197)